### PR TITLE
Add Find Doctor Command

### LINKED ISSUE #41: Find Command to search for Doctor

### PR DESCRIPTION
**Pull request by aldentantan**
Added a command to filter users by their assigned doctors. Closes #41

### CODE REVIEW

**Comment by aldentantan** [comment]
Can merge after I address Codecov

**Comment by aldentantan** [comment]
Closes #76

**Review by saajidshaik02 — CHANGES_REQUESTED** [review]
Please see below

File: docs/UserGuide.md [hunk 1]
```diff
@@ -138,6 +138,24 @@ Examples:
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
   ![result for 'find alex david'](images/findAlexDavidResult.png)
 
+### Locating persons by name: `finddoc`
+
+Finds persons by checking if their doctor's names contain any of the provided keywords.
+
```
Comment (saajidshaik02) [inline docs/UserGuide.md:144]: Good use of image for the userguide!

**Review by saajidshaik02 — CHANGES_REQUESTED** [review]
Please see below

File: docs/UserGuide.md [hunk 2]
```diff
@@ -138,6 +138,24 @@ Examples:
...
+* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* Only the doctor's name is searched.
+* Names will match if the keyword is found in any part of the doctor's name. For example, `Ha` will match `Hans`
+* Persons matching at least one keyword in their doctor's name will be returned (i.e. `OR` search).
+  e.g. `Hans Bo` will return persons whose doctors are `Hans Gruber`, `Bo Yang`
+
+Examples:
+* `finddoc John` returns persons with doctors `john` and `John Doe`
+* `finddoc tan ed` returns persons with doctors `Tan Wei Ming`, `Ed Sheeran`<br>
+  ![result for 'finddoc tan ed'](images/finddoc-tan-ed.png)
```
Comment (kellywsq03) [inline docs/UserGuide.md:157]: You may want to update this image when the UI for the EmergencyContacts is updated!

File: src/main/java/seedu/address/model/person/DoctorNameContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+public class DoctorNameContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public DoctorNameContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
```
[the remark this answers is not in the document]
Comment (kellywsq03) [reply src/main/java/seedu/address/model/person/DoctorNameContainsKeywordsPredicate.java:21]: Duper succinct code! Brought tears to my eyes :)

File: src/test/java/seedu/address/logic/parser/FindDoctorCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Test
+    public void parse_validArgs_returnsFindDoctorCommand() {
+        // no leading and trailing whitespaces
+        FindDoctorCommand expectedFindDoctorCommand =
+                new FindDoctorCommand(new DoctorNameContainsKeywordsPredicate(Arrays.asList("Alice", "Bob")));
+        assertParseSuccess(parser, "Alice Bob", expectedFindDoctorCommand);
+
+        // multiple whitespaces between keywords
```
Comment (kellywsq03) [inline src/test/java/seedu/address/logic/parser/FindDoctorCommandParserTest.java:31]: Helpful comments for developers!

**Review by kellywsq03 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 2 above)
Comment (aldentantan) [reply docs/UserGuide.md:157]: Will remember to update once all the PR's are merged, thanks!

**Review by kellywsq03 — APPROVED** [review]

**Review by saajidshaik02 — APPROVED** [review]
LGTM!

**Pull request merged**